composer/core/state.py

```diff
@@ -579,26 +586,46 @@ def _apply_required_algorithms(
                             textwrap.dedent(
                                 f"required_on_load algorithm {serialized_value['repr']} was enabled when training the "
                                 f"loaded checkpoint but is now specified in the following forms: {', '.join(state_algos[type(algo)])}."
-                                'Potential paarameter discrepencies for this required_on_load algorithm may lead to '
+                                'Potential parameter discrepancies for this required_on_load algorithm may lead to '
                                 'unexpected behavior, including failing to load weights for some layers.'))
                     elif type(algo) not in state_algos:
-                        missing_algorithms.append((algo, serialized_value['repr']))
+                        missing_algos.add(algo)
+                        missing_algo_names.append(algo_name)
+                        missing_algo_reprs.append(serialized_value['repr'])
+                        self.algorithms.append(algo)
+
+        # Reorder algorithms based on algorithm_passes from engine
+        algorithms = self.algorithms
+        if algorithm_passes is not None:
+            for passes in algorithm_passes:
+                algorithms = passes(algorithms, Event.INIT)
+        # Raise ValueError if algorithm_passes order any checkpoint algorithm is ordered before an
```

Comment:
"order any checkpoint algorithm is ordered before" -> "algorithm_passes order any checkpoint algorithm before"